### ISSUE TITLE
[import/no-extraneous-dependencies] 2.20.1 regression - false negatives

### ISSUE THREAD

**Issue by EvgenyOrekhov** [benmosher/eslint-plugin-import#1646]
The rule stopped working for me in 2.20.1.

Rule config:

```text
    "import/no-extraneous-dependencies": [
      "error",
      {
        "devDependencies": ["server/**", "tests/**"],
        "optionalDependencies": false,
        "peerDependencies": false
      }
    ],
```

`devDependencies` in `package.json`:

```text
...
"ember-qunit": "^4.4.1",
"qunit-dom": "^1.0.0",
...
```

In my tests:

```js
import { module, test } from "qunit";
```

Actual result: no warnings from `import/no-extraneous-dependencies`.

Expected result: there should be a warning from `import/no-extraneous-dependencies` saying that I have an extraneous `qunit` dependency.

Environment info:

OS: Win 10 x64
Node: v12.15.0
npm: 6.13.4

**Comment by ljharb** [comment]
In what filename is that test example? I'm assuming it's in `tests/**`.

**Comment by EvgenyOrekhov** [comment]
@ljharb Can't say the exact name right now, but yes, it's inside my `tests` folder.

**Comment by EvgenyOrekhov** [comment]
I'm getting that error for all my test files.

**Comment by ljharb** [comment]
And to confirm, in 2.20.0, it worked?

https://github.com/benmosher/eslint-plugin-import/compare/v2.20.0...v2.20.1 does not suggest any obvious cause.

Can you check your version of `eslint-module-utils`?

**Comment by EvgenyOrekhov** [comment]
@ljharb Yes, this is the only change in my `package-lock.json` after which the problem occurs:

```diff
    "eslint-plugin-import": {
-      "version": "2.20.0",
-      "resolved": "https://registry.npmjs.org/eslint-plugin-import/-/eslint-plugin-import-2.20.0.tgz",
-      "integrity": "sha512-NK42oA0mUc8Ngn4kONOPsPB1XhbUvNHqF+g307dPV28aknPoiNnKLFd9em4nkswwepdF5ouieqv5Th/63U7YJQ==",
+      "version": "2.20.1",
+      "resolved": "https://registry.npmjs.org/eslint-plugin-import/-/eslint-plugin-import-2.20.1.tgz",
+      "integrity": "sha512-qQHgFOTjguR+LnYRoToeZWT62XM55MBVXObHM6SKFd1VzDcX/vqT1kAz8ssqigh5eMj8qXcRoXXGZpPP6RfdCw==",
      "dev": true,
      "requires": {
```

`eslint-module-utils` version is the same as before: 2.5.2.

```text
$ npm ls eslint-module-utils
my-app@0.0.0 C:\my-app
`-- eslint-plugin-import@2.20.1
  `-- eslint-module-utils@2.5.2
```

**Comment by EvgenyOrekhov** [comment]
Added environment info to the bottom of the issue description, if that's any help.

**Comment by cattermo** [comment]
I also have problem with 2.20.1 version. In my case I use eslint-import-resolver-webpack to resolve aliases. From version 2.20.1 I get the import/no-extraneous-dependencies error for all my imports of react event though it is aliased in webpack.config.js (this is working in version 2.20.0).

OSX 10.14.6 
Node v10.17.0
Npm 6.11.3

**Comment by EvgenyOrekhov** [comment]
The issue is no longer reproducible on 2.22.1.
